internal/functions/efgs/redis/redis.go

```diff
@@ -0,0 +1,83 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+	"github.com/covid19cz/erouska-backend/internal/logging"
+	redisclient "github.com/go-redis/redis/v8"
+	"os"
+	"sync"
+	"time"
+)
+
+var redisClient *Connection
+var ctx context.Context
+
+type lazyConnection func() *redisclient.Client
+
+//Connection Contains lazy Redis connection
+type Connection struct {
+	inner lazyConnection
+}
+
+func init() {
+	ctx = context.Background()
+
+	connect := func() *redisclient.Client {
+		logger := logging.FromContext(ctx).Named("efgs.redis.connect")
+
+		logger.Debug("Connecting to EFGS Redis")
+
+		addr, ok := os.LookupEnv("EFGS_REDIS_ADDR")
+		if !ok {
+			panic("EFGS_REDIS_ADDR env missing")
+		}
+
+		client := redisclient.NewClient(&redisclient.Options{
+			Addr: addr,
+			DB:   0,
+		})
```

Comment:
Is this going to run locally when running unit tests against it?

Reply:
I'm not sure what do you mean. Currently, there's no `MockClient` or whatever as, well, there are no unit tests. The one who will write them can add the mock easily... or, of course, I can do it now, if you insist :wink:

Reply:
even without unit tests - if I try to run this locally with something like EFGS_REDIS_ADDR=NOOP, it will panic. These init() functions that expect you have to have everything up and running are evil :D

Reply:
unit tests would be perfect but my point was to at least make it like with firestore / other dependencies that can be skipped using something like setting an ENV var to NOOP